From 5395ccff4478d42fab7f2bd6ba7f7188bec09ddf Mon Sep 17 00:00:00 2001
From: Yash Shinde <Yash.Shinde@windriver.com>
Date: Fri, 5 Apr 2024 08:24:36 -0700
Subject: [PATCH] tests: Skip 2 qemu tests that can hang in oe-selftest

qemumips and qemuppc were leaving stale processes behind after
running glibc oe-selftest. During analysis, it was found that
it was due to "tst-scm_rights" and "tst-scm_rights-time64" tests.
Disable them so that there are no stale processes left behind.

[YOCTO #15423]
https://bugzilla.yoctoproject.org/show_bug.cgi?id=15423

Upstream-Status: Inappropriate [oe-core specific]

Signed-off-by: Yash Shinde <Yash.Shinde@windriver.com>
Signed-off-by: Khem Raj <raj.khem@gmail.com>
---
 sysdeps/unix/sysv/linux/Makefile | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/sysdeps/unix/sysv/linux/Makefile b/sysdeps/unix/sysv/linux/Makefile
index 8a755293b3..22dafcaad1 100644
--- a/sysdeps/unix/sysv/linux/Makefile
+++ b/sysdeps/unix/sysv/linux/Makefile
@@ -228,7 +228,6 @@ tests += \
   tst-rlimit-infinity \
   tst-sched-affinity-inheritance \
   tst-sched_setattr \
-  tst-scm_rights \
   tst-sigtimedwait \
   tst-sync_file_range \
   tst-syscall-restart \
@@ -240,6 +239,8 @@ tests += \
   tst-timerfd \
   tst-ttyname-direct \
   tst-ttyname-namespace \
+  # Skip this test to avoid stale qemu process
+  # tst-scm_rights \
   # tests
 
 # process_madvise requires CAP_SYS_ADMIN.
@@ -277,9 +278,10 @@ tests-time64 += \
   tst-ntp_gettimex-time64 \
   tst-ppoll-time64 \
   tst-prctl-time64 \
-  tst-scm_rights-time64 \
   tst-sigtimedwait-time64 \
   tst-timerfd-time64 \
+  # Skip this test to avoid stale qemu process
+  # tst-scm_rights-time64 \
   # tests-time64
 
 tests-clone-internal = \
